services/dedup/dedup.go

```diff
@@ -111,25 +111,17 @@ func (d *DedupHandleT) openBadger() {
 
 	opts := badger.
 		DefaultOptions(d.path).
-		WithTruncate(true).
 		WithLogger(d.logger).
-		// Disable compression - Set options.Compression = options.None.
-		// This means we won’t allocate memory for decompression
-		// (this can be a lot in case of ZSTD decompression).
-		// In our case, compression is not useful since we are storing messageIDs with high entropy.
 		WithCompression(options.None)
+	// Disable compression - Set options.Compression = options.None.
+	// This means we won’t allocate memory for decompression
+	// (this can be a lot in case of ZSTD decompression).
+	// In our case, compression is not useful since we are storing messageIDs with high entropy.
 
 	if memOptimized {
-		// Memory usage optimizations:
-		// Inspired by https://github.com/dgraph-io/badger/issues/1304#issuecomment-630078745
-		// With modifications to ensure no performance degradation for dedup.
-		opts.TableLoadingMode = options.FileIO
-		opts.ValueLogLoadingMode = options.FileIO
 		opts.NumMemtables = 3
-		opts.MaxTableSize = 16 << 20
 		opts.NumLevelZeroTables = 1
 		opts.NumLevelZeroTablesStall = 2
-		opts.KeepL0InMemory = false
```

Comment:
Also, over here a lot of variable are no more present. Can you explain, how these things are being handled even if we don't sent them?